librz/il/rzil_vm.c

```diff
@@ -0,0 +1,724 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_il/rzil_vm.h>
+
+/**
+ * Create A new variable in VM
+ * \param vm RzILVM, pointer to VM
+ * \param name string, name of this variable
+ * \return var RzILVar, pointer to the new variable in VM
+ */
+RZ_API RzILVar rz_il_vm_create_variable(RzILVM vm, char *name) {
+	if (vm->var_count >= RZ_IL_VM_MAX_VAR) {
+		printf("No More Vars here\n");
+		return NULL;
+	}
+
+	// create , store, update count
+	RzILVar var = rz_il_new_variable(name);
+	vm->vm_global_variable_list[vm->var_count] = var;
+	vm->var_count += 1;
+
+	return var;
+}
+
+/**
+ * Create a new value in VM
+ * \param vm RzILVM, pointer to VM
+ * \param type RZIL_VAR_TYPE, enum to specify the type of this value
+ * \return val RzILVal, pointer to the new value in VM
+ */
+RZ_API RzILVal rz_il_vm_create_value(RzILVM vm, RZIL_VAR_TYPE type) {
+	if (vm->val_count >= RZ_IL_VM_MAX_VAL) {
+		printf("No More Values\n");
+		return NULL;
+	}
+
+	RzILVal val = rz_il_new_value();
+	val->type = type;
+
+	rz_il_add_to_bag(vm->vm_global_value_set, val);
+	return val;
+}
+
+/**
+ * Add a register in VM (create a variable and value, and then bind value to variable)
+ * \param vm RzILVM, pointer to this vm
+ * \param name string, the name of register
+ * \param length int, width of register
+ */
+RZ_API void rz_il_vm_add_reg(RzILVM vm, char *name, int length) {
```

Comment:
Same

Reply:
Also do a `rz_return_if_fail(lenght > 0)`. Probably also make sense to make `length` `size_t` or `unsigned int`.